### PR TITLE
Makefile: Add clang formatting target

### LINKED ISSUE #566: [NEW] Update CONTRIBUTING.md and improve developer experience around clang-format
We could also add a make target like `make format` that just formats the code appropriately, which can be run if CI fails or before committing.

_Originally posted by @bjosv in https://github.com/valkey-io/valkey/issues/538#issuecomment-2134473903_

### PR DESCRIPTION
**Pull request by hahnandrew**
- [x] Add a make target `format` to format files with clang
- [x] Update CONTRIBUTING.md

Fixes #566 
Addresses @bjosv in https://github.com/valkey-io/valkey/pull/538#issuecomment-2134473903

### CODE REVIEW

**Comment by hahnandrew** [comment]
Can I get your :eyes: on this PR @PingXie?

File: src/Makefile [hunk 1]
```diff
@@ -639,3 +639,25 @@ uninstall:
...
 	$(call MAYBE_UNINSTALL_REDIS_SYMLINK,$(INSTALL_BIN),$(ENGINE_CHECK_AOF_NAME))
 	$(call MAYBE_UNINSTALL_REDIS_SYMLINK,$(INSTALL_BIN),$(ENGINE_SENTINEL_NAME))
+
+format:
+	@if ! command -v clang-format >/dev/null 2>&1; then \
+		echo "Error: clang-format is required to run this command."; \
+		exit 1; \
+	fi
+	@CLANG_FORMAT_VERSION=$$(clang-format --version | awk '{print $$3}' | cut -d. -f1); \
+	if [ $$CLANG_FORMAT_VERSION -lt 18 ]; then \
```
Comment (hahnandrew) [inline src/Makefile:649]: https://github.com/valkey-io/valkey/issues/649

Comment (PingXie) [inline src/Makefile:649]: I  think we will have to stick to clang 18 for now. I have run into some behavior differences between 18 and 19.

File: src/Makefile [hunk 2]
```diff
@@ -639,3 +639,25 @@ uninstall:
...
+		echo "Error: clang-format version 18 or higher is required."; \
+		echo "Your version: $$(clang-format --version)"; \
+		echo "Please upgrade clang-format to continue."; \
+		exit 1; \
+	fi
+	@echo "Formatting code.."
+	@trap 'kill $$!' EXIT; \
+	{ while true; do for s in / - \\ \|; do printf "\r%s" $$s; sleep 0.1; done; done } & \
+	{ \
+		find . \( -name '*.c' -o -name '*.h' \) -type f | xargs clang-format -i -style=file; \
```
Comment (PingXie) [inline src/Makefile:659]: `find .` might be problematic as `make` can be invoked from both `src` or its parent directory. In either case, we would want to format just the files under `src`

**Review by PingXie — COMMENTED** [review]
Thanks for the PR, @hahnandrew!

Sorry I didn't know that git already has full clang-format [integration](https://ortogonal.github.io/cpp/git-clang-format/) when I filed the original issue. I am not sure if there is much value in churning Makefile further.  `git clang-format` has been working pretty well for my use cases.

**Comment by hahnandrew** [comment]
Closed in favor of [`git clang-format`](https://github.com/valkey-io/valkey/issues/566#issuecomment-2298094251)

**Pull request closed without merging**